src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/MinMax.cs

```diff
@@ -1,21 +1,81 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+using System;
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
 using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Syntax.Nodes;
 using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
 
 namespace Microsoft.PowerFx.Core.Texl.Builtins
 {
     // Min(arg1:n, arg2:n, ..., argN:n)
     // Max(arg1:n, arg2:n, ..., argN:n)
     // Corresponding Excel functions: Min, Max
-    internal sealed class MinMaxFunction : StatisticalFunction
+    internal sealed class MinMaxFunction : BuiltinFunction
     {
         public override bool SupportsParamCoercion => true;
 
+        public override bool IsSelfContained => true;
+
         public MinMaxFunction(bool isMin)
-            : base(isMin ? "Min" : "Max", isMin ? TexlStrings.AboutMin : TexlStrings.AboutMax, FunctionCategories.MathAndStat)
+            : base(isMin ? "Min" : "Max", isMin ? TexlStrings.AboutMin : TexlStrings.AboutMax, FunctionCategories.MathAndStat, DType.Number, 0, 1, int.MaxValue, DType.Number)
+        {
+        }
+
+        public override bool CheckInvocation(TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            nodeToCoercedTypeMap = null;
+            var fArgsValid = true;
+            returnType = argTypes[0];
+
+            // If there are elements of different types OR if the elements are not a Date/Time/DateTime, coerce to numeric.
+            if (!Array.TrueForAll(argTypes, element => element == argTypes[0]) || !Array.Exists(argTypes, element => element == DType.Date || element == DType.DateTime || element == DType.Time))
```

Comment:
It would be good to add some tests for expressions that don't compile. Things like

    Min({a:1}, {a:2}, {a:3})
    Max(1, 2, {a:3})
    Min([1,2,3], [1])

This is very similar to the previous one, but it should work, so we should add as a test:

    >> Min([1, 2, 3], 4) // This uses the `Min(<table>, <expression>)` overload, and the expression is a constant
    4

Reply:
Thanks, tests added!